src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -966,31 +966,49 @@ private async Task CreateExecutableAsync(AppResource er, ILogger resourceLogger,
+        // provided by the launch profile.
+        // https://github.com/dotnet/aspire/blob/main/docs/specs/IDE-execution.md#launch-profile-processing-project-launch-configuration
+        spec.Args = null;
+
         // An executable can be restarted so args must be reset to an empty state.
         // After resetting, first apply any dotnet project related args, e.g. configuration, and then add args from the model resource.
-        spec.Args = [];
-        if (er.DcpResource.TryGetAnnotationAsObjectList<string>(CustomResource.ResourceProjectArgsAnnotation, out var projectArgs))
+        if (er.DcpResource.TryGetAnnotationAsObjectList<string>(CustomResource.ResourceProjectArgsAnnotation, out var projectArgs) && projectArgs.Count > 0)
         {
+            spec.Args ??= [];
             spec.Args.AddRange(projectArgs);
         }
 
+        (var args, var failedToApplyArgs) = await BuildArgsAsync(resourceLogger, er.ModelResource, cancellationToken).ConfigureAwait(false);
+
         var launchArgs = new List<(string Value, bool IsSensitive, bool AnnotationOnly)>();
 
         // If the executable is a project then include any command line args from the launch profile.
         if (er.ModelResource is ProjectResource project)
         {
-            // When the .NET project is launched from an IDE the launch profile args are automatically added.
-            // We still want to display the args in the dashboard so only add them to the custom arg annotations.
-            var annotationOnly = spec.ExecutionType == ExecutionType.IDE;
+            // Args in the launch profile is used when:
+            // 1. The project is run as an executable. Launch profile args are combined with app host supplied args.
+            // 2. The project is run by the IDE and no app host args are specified.
+            if (spec.ExecutionType == ExecutionType.Process || (spec.ExecutionType == ExecutionType.IDE && args.Count == 0))
+            {
+                // When the .NET project is launched from an IDE the launch profile args are automatically added.
+                // We still want to display the args in the dashboard so only add them to the custom arg annotations.
+                var annotationOnly = spec.ExecutionType == ExecutionType.IDE;
 
-            var launchProfileArgs = GetLaunchProfileArgs(project.GetEffectiveLaunchProfile()?.LaunchProfile);
-            launchArgs.AddRange(launchProfileArgs.Select(a => (a, isSensitive: false, annotationOnly)));
+                var launchProfileArgs = GetLaunchProfileArgs(project.GetEffectiveLaunchProfile()?.LaunchProfile);
+                launchArgs.AddRange(launchProfileArgs.Select(a => (a, isSensitive: false, annotationOnly)));
+            }
         }
 
-        (var args, var failedToApplyArgs) = await BuildArgsAsync(resourceLogger, er.ModelResource, cancellationToken).ConfigureAwait(false);
         launchArgs.AddRange(args.Select(a => (a.Value, a.IsSensitive, annotationOnly: false)));
```

Comment:
I know this won't actually do anything if line 992 is true, and so the effective launch args will be correct, but did a double take - could a comment or a condition be added to clarify?